x-pack/platform/plugins/shared/event_log/server/es/cluster_client_adapter.ts

```diff
@@ -135,6 +157,26 @@ export class ClusterClientAdapter<TDoc extends { body: AliasAny; index: string }
     await this.docsBufferedFlushed;
   }
 
+  public async updateDocument(doc: Required<TDoc>) {
+    const esClient = await this.elasticsearchClientPromise;
+    try {
+      if (!doc.internalFields) {
+        throw new Error('Internal fields are required');
+      }
+      await esClient.update({
+        doc: doc.body,
+        id: doc.internalFields._id,
+        index: doc.internalFields._index,
+        if_primary_term: doc.internalFields._primary_term,
+        if_seq_no: doc.internalFields._seq_no,
+        refresh: 'wait_for',
```

Comment:
If we're doing a lot of consecutive updates and waiting for refresh each time, this could quickly balloon the time we're spending updating these docs. I believe the refresh rate in serverless is `5s` and using `wait_for` does not force a refresh. If we're expecting to do a lot of updates, we should consider doing a bulk request so we're not waiting for consecutive refreshes. We should also consider whether this is a candidate for `refresh: true` which does have performance implications in Elasticsearch if we're constantly forcing refreshes on a index.